samples/VanillaJSTestApp/index_blankPageRedirectUri.html

```diff
@@ -0,0 +1,200 @@
+            }
+        });
+    }
+
+    // Call Graph to fetch data
+    function callMSGraph(theUrl, accessToken, callback) {
+        var xmlHttp = new XMLHttpRequest();
+        xmlHttp.onreadystatechange = function () {
+            if (this.readyState == 4 && this.status == 200)
+                callback(JSON.parse(this.responseText));
+        }
+        xmlHttp.open("GET", theUrl, true); // true for asynchronous
+        xmlHttp.setRequestHeader('Authorization', 'Bearer ' + accessToken);
+        xmlHttp.send();
+    }
+
+    function graphAPICallback(data) {
+        document.getElementById("json").innerHTML = JSON.stringify(data, null, 2);
+    }
+
+    function showWelcomeMessage() {
+        var divWelcome = document.getElementById('WelcomeMessage');
+        divWelcome.innerHTML = 'Welcome ' + myMSALObj.getAccount().userName + " to Microsoft Graph API";
+        var loginbutton = document.getElementById('SignIn');
+        loginbutton.innerHTML = 'Sign Out';
+        loginbutton.setAttribute('onclick', 'signOut();');
+    }
+
+    function readMail() {
+        acquireTokenPopupAndCallMSGraph(graphConfig.graphMailEndpoint, tokenRequest);
+    }
+
+    // signout
+    function signOut() {
+        myMSALObj.logout();
+    }
+
+   // This function can be removed if you do not need to support IE
+   function acquireTokenRedirectAndCallMSGraph(endpoint, request) {
+        //Call acquireTokenSilent (iframe) to obtain a token for Microsoft Graph
+        myMSALObj.acquireTokenSilent(request).then(function (tokenResponse) {
+            callMSGraph(endpoint, tokenResponse.accessToken, graphAPICallback);
+        }).catch(function (error) {
+            console.log("error is: "+ error);
+            console.log("stack:" + error.stack);
+            //Call acquireTokenRedirect in case of acquireToken Failure
+            if (requiresInteraction(error.errorCode)) {
+                myMSALObj.acquireTokenRedirect(request);
+            }
+        });
+    }
+
+    // redirect call back
+    function authRedirectCallBack(error, response) {
+        if (error) {
+            console.log(error);
+        } else {
+            if (response.tokenType === "id_token") {
+                showWelcomeMessage();
+                acquireTokenRedirectAndCallMSGraph(graphConfig.graphMeEndpoint, loginRequest);
+            } else if (response.tokenType === "access_token") {
+                callMSGraph(graphConfig.graphMeEndpoint, response.accessToken, graphAPICallback);
+            } else {
+                console.log("token type is:" + response.tokenType);
+            }
+        }
+    }
+
+    // utils to handle standard error set that would need user interaction
+    function requiresInteraction(errorMessage) {
+        if (!errorMessage || !errorMessage.length) {
+            return false;
+        }
+
+        console.log("requiresinteraction is:" + errorMessage );
+        return errorMessage.indexOf("consent_required") !== -1 ||
```

Comment:
Can we upgrade this to use the built-in static function?

Reply:
yes, I will add changes with a separate PR soon. Can we delink this with the release effort?